src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml

```diff
@@ -113,108 +120,118 @@
-                    <Run Text="{Binding CheckoutDisabledMessage}"/>
-                    <TextBlock.Style>
-                        <Style TargetType="TextBlock" BasedOn="{StaticResource CheckoutErrorMessage}">
-                            <Style.Triggers>
-                                <DataTrigger Binding="{Binding CheckoutDisabledMessage}" Value="{x:Null}">
-                                    <Setter Property="Visibility" Value="Collapsed" />
-                                </DataTrigger>
-                            </Style.Triggers>
-                        </Style>
-                    </TextBlock.Style>
-                </TextBlock>
-
-                <!-- Git operation error message -->
-                <TextBox Foreground="Red" 
-                         Margin="0 4"
-                         Text="{Binding OperationError}"
-                         TextWrapping="Wrap"
-                         Style="{StaticResource FlatReadOnlyTextBox}"
-                         Visibility="{Binding OperationError, Converter={ui:NullToVisibilityConverter}}"/>
+                <StackPanel Orientation="Horizontal">
+                    <TextBlock Opacity="0.5" VerticalAlignment="Center"
+                               Text="{Binding Model.UpdatedAt, StringFormat={}updated {0}, Converter={ui:DurationToStringConverter}, Mode=OneWay}"/>
+
+                    <Rectangle Margin="5" Width="1" Height="12" VerticalAlignment="Center" Style="{DynamicResource Separator}" />
+
+
+                    <!-- Actions area (checkout, pull, push) -->
+
+                    <!-- Checkout pull request button -->
+                    <ui:GitHubActionLink Command="{Binding Checkout}"
+                                         Content="{Binding CheckoutState.Caption}"
+                                         VerticalAlignment="Center"
+                                         TextWrapping="Wrap"
+                                         Visibility="{Binding CheckoutState, Converter={ui:NullToVisibilityConverter}}"
+                                         ToolTip="{Binding CheckoutState.DisabledMessage}"
+                                         ToolTipService.ShowOnDisabled="True"/>
+
+                    <!-- Pull/push buttons -->
+                    <StackPanel Orientation="Horizontal" VerticalAlignment="Center"
+                                Visibility="{Binding UpdateState.UpToDate, FallbackValue=Collapsed, Converter={ui:BooleanToInverseVisibilityConverter}}">
+                        <ui:OcticonImage Icon="arrow_down"/>
+                        <TextBlock Text="{Binding UpdateState.CommitsBehind}" VerticalAlignment="Center"/>
+                        <ui:GitHubActionLink Content="Pull"
+                                             Command="{Binding Pull}"
+                                             Margin="4,0"
+                                             ToolTip="{Binding UpdateState.PullDisabledMessage}"
```

Comment:
Would it make sense to have a tooltip that says something like "Pull changes from the remote branch foo-branch to your local branch?" If so what's the best way to do this so that it also works with this disabled message?